Extension/src/LanguageServer/cppBuildTaskProvider.ts

```diff
@@ -387,11 +394,29 @@ class CustomBuildTaskTerminal implements Pseudoterminal {
         const exePath: string | undefined = util.resolveVariables(util.findExePathInArgs(this.args));
         util.createDirIfNotExistsSync(exePath);
 
-        this.args.forEach((value, index) => {
-            value = util.quoteArgument(util.resolveVariables(value));
-            activeCommand = activeCommand + " " + value;
-            this.args[index] = value;
-        });
+        if (this.fragments.length > 0 && this.args.length === 0) {
```

Comment:
Could this be simplified to? :

```
        this.args.forEach((value, index) => {
            value = util.quoteArgument(util.resolveVariables(value));
            activeCommand = activeCommand + " " + value;
            this.args[index] = value;
        });
        this.fragments.forEach((value) => {
            activeCommand = activeCommand + " " + value;
            this.args.push(value);
        });
```

Or, as I've suggested before, change `cp.spawn` to `cp.exec`, so we can just pass it `activeCommand`, which we're already building for logging purposes.  Then you could omit all of the edits of `this.args`, reducing that code to:

```
        this.args.forEach((value, index) => {
            value = util.quoteArgument(util.resolveVariables(value));
            activeCommand = activeCommand + " " + value;
        });
        this.fragments.forEach((value) => {
            activeCommand = activeCommand + " " + value;
        });
```